packages/grid/src/KeyHandler.ts

```diff
@@ -6,16 +6,26 @@
  */
 
 // eslint-disable-next-line import/no-cycle
+import { KeyboardEvent } from 'react';
 import Grid from './Grid';
 
-// True if consumed and to stop event propagation/prevent default, false if not consumed.
-// OR an object if consumed with boolean properties to control whether to stopPropagation/preventDefault
-export type KeyHandlerResponse =
-  | boolean
-  | { stopPropagation?: boolean; preventDefault?: boolean };
+/**
+ * An options object can be returned as the result to control
+ * if event.stopPropagation() and event.preventDefault() should be called
+ */
+export type KeyHandlerResultOptions = {
+  stopPropagation?: boolean;
+  preventDefault?: boolean;
+};
+
+/**
+ * Result from the key handler for an event.
+ * Return `false` to pass the event on, otherwise the event is consumed.
+ */
+export type KeyHandlerResult = boolean | KeyHandlerResultOptions;
```

Comment:
I had a separate type because in Grid need to cast it to that other object type with `stopPropagation` and `preventDefault`, seemed cleaner. 
There are some cases we return either false or true, or the object.
I've updated the comment but I'm not sure updating the name would make much of a difference?